src/allencell_ml_segmenter/main/viewer.py

```diff
@@ -153,33 +153,16 @@ def insert_threshold(
         If the layer does not exist, it will be added to the viewer in the correct place (on top of the original segmentation image:
         index_of_segmentation + 1 in the LayerList)
 
-        :param layer_name: name of layer to insert. Will replace if one exists, will create one in a new position if needed.
+        :param layer: layer to replace.
         :param image: image to insert
         :param remove_seg_layers: boolean indicating if the layer that is being thresholded is a segmentation layer, and should be removed from the layer once it is updated with the threshold.
         """
-        layer_to_insert = self._get_layer_by_name(f"[threshold] {layer_name}")
-        if layer_to_insert is None:
-            # No thresholding exists, so we add it to the correct place in the viewer
-            layerlist = self.viewer.layers
-
-            # check if the original segementation layer is currently in the viewer, if so, remove later after
-            # thresholding is applied
-            seg_layer_og: Optional[Layer] = None
-            if remove_seg_layers:
-                seg_layer_og = self._get_layer_by_name(layer_name)
-
-            # figure out where to insert the new thresholded layer (on top of the original segmentation image)
-            layerlist_pos = layerlist.index(layer_name)
-            labels_created = Labels(image, name=f"[threshold] {layer_name}")
-            layerlist.insert(layerlist_pos + 1, labels_created)
-
-            # remove the original segmentation layer if it exists
-            if seg_layer_og:
-                layerlist.remove(seg_layer_og)
-        else:
-            # Thresholding already exists so just update the existing one in the viewer.
-            layer_to_insert.data = image
-            layer_to_insert.refresh()
+        # if threshold has not been previously applied, update name
+        if "threshold_applied" not in layer.metadata:
```

Comment:
We still have threshold specific code being handled in our abstracted Viewer - can this be handled in the thresholding layer?